apps/launch/components/MyStakingContracts/FieldConfig.tsx

```diff
@@ -0,0 +1,313 @@
+    description={FieldConfig.maxNumInactivityPeriods.desc}
+  />
+);
+
+export const LivenessPeriodLabel = () => (
+  <TextWithTooltip
+    text={FieldConfig.livenessPeriod.name}
+    description={FieldConfig.livenessPeriod.desc}
+  />
+);
+
+export const TimeForEmissionsLabel = () => (
+  <TextWithTooltip
+    text={FieldConfig.timeForEmissions.name}
+    description={FieldConfig.timeForEmissions.desc}
+  />
+);
+
+export const AgentInstancesLabel = () => (
+  <TextWithTooltip
+    text={FieldConfig.numAgentInstances.name}
+    description={FieldConfig.numAgentInstances.desc}
+  />
+);
+
+export const AgentIdsLabel = () => (
+  <TextWithTooltip text={FieldConfig.agentIds.name} description={FieldConfig.agentIds.desc} />
+);
+
+export const MultisigThresholdLabel = () => (
+  <TextWithTooltip text={FieldConfig.threshold.name} description={FieldConfig.threshold.desc} />
+);
+
+export const ServiceConfigHashLabel = () => (
+  <TextWithTooltip text={FieldConfig.configHash.name} description={FieldConfig.configHash.desc} />
+);
+
+export const ActivityCheckerAddressLabel = () => (
+  <TextWithTooltip
+    text={FieldConfig.activityChecker.name}
+    description={FieldConfig.activityChecker.desc}
+  />
+);
+
+/** ******* RULES ******* */
+type StakingDepositRules = { [K in keyof FormValues]: { rules: Rule[] | undefined } };
+export const useCreateStakingContractRules = (): StakingDepositRules => {
+  const { data: numServicesLimit } = useNumServicesLimit();
+  const { data: minStakingDepositLimit } = useMinStakingDepositLimit();
+  const { data: timeForEmissionsLimit } = useTimeForEmissionsLimit();
+
+  return {
+    contractName: { rules: getGenericFieldRules(FieldConfig.contractName.name) },
+    description: { rules: getGenericFieldRules(FieldConfig.description.name) },
+    maxNumServices: {
+      rules: [
+        ...getGenericFieldRules(FieldConfig.maxNumServices.name),
+        {
+          type: 'number',
+          min: 1,
+          max: numServicesLimit,
+          message: `Maximum number of staked agents must be at least 1 and at most ${numServicesLimit}`,
+        },
+      ],
+    },
+    rewardsPerSecond: { rules: getGenericFieldRules(FieldConfig.rewardsPerSecond.name) },
+    minStakingDeposit: {
+      rules: [
+        ...getGenericFieldRules(FieldConfig.minStakingDeposit.name),
+        {
+          type: 'number',
+          min: 1,
+          max: minStakingDepositLimit,
+          message: `Minimum service staking deposit, OLAS must be at least 1 and at most ${minStakingDepositLimit}`,
+        },
+      ],
+    },
+    minNumStakingPeriods: { rules: getGenericFieldRules(FieldConfig.minNumStakingPeriods.name) },
+    maxNumInactivityPeriods: {
+      rules: getGenericFieldRules(FieldConfig.maxNumInactivityPeriods.name),
+    },
+    livenessPeriod: {
+      rules: [
+        ...getGenericFieldRules(FieldConfig.livenessPeriod.name),
+        {
+          type: 'number',
+          min: 86400,
+          max: 86400 * 30,
```

Comment:
I don't see any min/max rules for this field anywhere, could you please tell me where you get them from?